ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -127,11 +98,34 @@ constructor(
       .doOnSubscribe { Timber.d("Loading survey $id") }
       .doOnError { err -> Timber.d(err, "Error loading survey from remote") }
 
-  fun loadLastActiveSurvey() = activateSurvey(lastActiveSurveyId)
+  fun loadLastActiveSurvey() = activateSurvey(localValueStore.lastActiveSurveyId)
 
-  fun activateSurvey(surveyId: String) = selectSurveyEvent.onNext(surveyId)
+  fun activateSurvey(surveyId: String) {
+    // Do nothing if survey is already active.
+    if (surveyId == activeSurveyId) {
+      return
+    }
+    // Clear survey if id is empty.
+    if (surveyId.isEmpty()) {
+      clearActiveSurvey()
+      return
+    }
 
-  fun clearActiveSurvey() = selectSurveyEvent.onNext("")
+    externalScope.launch {
```

Comment:
I'm not convinced we should be launching anything in a repository. Shouldn't we write these as suspend functions an `launch` from a component that actually has a lifecycle? (viewmodel that calls the repo methods)

Reply:
Got it, that makes sense. Could we add some kotlindocs on the ApplicationScope explaining this? That its purpose is to launch coroutines that should have the same lifecycle as the app process?

Reply:
I know.. I have to say, coroutines and Hilt are a huge improvement , but there's still some complexity that needs to be managed.

Done!